app/services/appointment_notification_service.rb

```diff
@@ -7,7 +7,7 @@ def initialize(appointments:, days_overdue: 3, schedule_at:)
     @appointments = appointments
     @days_overdue = days_overdue
     @schedule_at = schedule_at
-    @communication_type = Communication.communication_types[:missed_visit_sms_reminder]
+    @communication_type = Communication.communication_types[:missed_visit_whatsapp_reminder]
```

Comment:
I've tried to do this but may have missed something. Please have a look :)

Reply:
Hmm, the problem is this will always be hardcoded to whatsapp for the `communication_type`, right? So this piece specifically needs to take into account the feature flag as well.

Reply:
I think the communication type needs to be the piece that checks the feature flag and switches appropriately...then the Worker uses the communication_type only to decide how to send reminders. In other words, this service should be the driver which looks at FeatureFlags, and the worker just consumes that data and does as its told.

Reply:
Yes, makes sense! 🙌

Reply:
That sounds nifty. I'd like to defer that coupling until _after_ we put the whatsapp -> sms fallback mechanism in place. Once that code's up, it may change our mind on whether coupling comms type to country is a good idea or not. What do you think @kitallis?

Reply:
Yeah, I think it's ok for now. However, even if we don't use country-specific configs right now, keep in mind that all our feature toggles are _already_ implicitly country-specific.

Reply:
With the caveat that they can be _more_ flexible than per-country. We can easily trial Whatsapp in Demo before launching in Prod this way :smile:

Reply:
We can do that with country configs too. But I understand your point, tying it to countries is a very specific _product-ish_ decision we'd be making.

My larger point is that, this naked `communication_type` string/sym is making decisions now, so we should think about encapsulating it, related point: https://github.com/simpledotorg/simple-server/pull/971#discussion_r428531475. This is a difficult refactor though, so we can talk about it separately.